### PR TITLE
improve error messages and docs

### LINKED ISSUE #5: STM32F407VGTx Do not have unwind info for the given address
Do we think its a configuration error or unsupported arch somewhere?

```bash
$ cargo run --release --example cycle_count
    Finished release [optimized] target(s) in 0.01s
     Running `probe-run --chip STM32F407VGTx target/thumbv7em-none-eabi/release/examples/cycle_count`
flashing program ..
DONE
resetting device
ticks: 16800059
stack backtrace:
   0: 0x0800132e - <unknown>
Error: Do not have unwind info for the given address.
```

### PR DESCRIPTION
**Pull request by japaric**
closes #5

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,6 +44,30 @@ Instead of `$CHIP` you'll need to write the name of your microcontroller.
...
+If you are using the `cortex-m-quickstart` template then this is already the case.
+If not check or add these lines to `Cargo.toml`.
+
+``` toml
+# Cargo.toml
+[profile.dev]
+debug = 1 # default is `true`; not needed if not already overridden
+
+[profile.release]
+debug = 1 # default is `false`; using `true` is also OK
```
Comment (jonas-schievink) [inline README.md:59]: Why do we need to enable this? Backtraces should work even without line info

Comment (japaric) [reply README.md:59]: the virtual unwinder (gimli) requires `.debug_frame` info to compute the stack pointer of the previous frame while walking up the stack. Last time I checked `.debug_frame` is not produced when `debug = false`.

Comment (jonas-schievink) [reply README.md:59]: Hmm, then how does it work as the backtrace implementation of libstd? That one works even without any debuginfo in the program.

Comment (japaric) [reply README.md:59]: Hmm, it seems `.debug_frame` is produced even with `debug = false`. Setting `debug = false` for `profile.dev` makes `probe-run` unwinder fail though; may be due to some other reason.

Comment (jonas-schievink) [reply README.md:59]: Okay, seems fine to merge this then. Maybe file an issue about the unwinder bug?

Comment (japaric) [reply README.md:59]: #9

**Pull request merged**